src/dns.ml

```diff
@@ -4726,29 +4727,29 @@ module Packet = struct
   let raw_error buf rcode =
     (* copy id from header, retain opcode, set rcode to ServFail
        if we receive a fragment < 12 bytes, it's not worth bothering *)
-    if Cstruct.length buf < 12 then
+    if String.length buf < 12 then
       None
     else
-      let query = Cstruct.get_uint8 buf 2 lsr 7 = 0 in
+      let query = String.get_uint8 buf 2 lsr 7 = 0 in
       if not query then (* never reply to an answer! *)
         None
       else
-        let hdr = Cstruct.create 12 in
+        let hdr = Bytes.make 12 '\000' in
```

Comment:
Here it seems that only a subpart is initialized (bytes 0-4 and 10-11 in the else alternative), so probably it's better to keep `Bytes.make` :)

Reply:
Sorry for the first comment, I just wanted to comment on each `Bytes.make`, and here I think it just added noise. This `Bytes.make` was replaced in https://github.com/mirage/ocaml-dns/pull/351/commits/68a823e7e7652bf4e6067d0ea3e6fea0906553fd among all that similar replacements. But, as I understand it, it shouldn't be since only a few bytes are defined.